packages/flutter_tools/lib/src/version.dart

```diff
@@ -217,52 +217,20 @@ class FlutterVersion {
     }
   }
 
-  /// The name of the temporary git remote used to check for the latest
-  /// available Flutter framework version.
-  ///
-  /// In the absence of bugs and crashes a Flutter developer should never see
-  /// this remote appear in their `git remote` list, but also if it happens to
-  /// persist we do the proper clean-up for extra robustness.
-  static const String _versionCheckRemote = '__flutter_version_check__';
-
-  /// The date of the latest framework commit in the remote repository.
+  /// The latest commit hash of the remote repository.
   ///
   /// Throws [VersionCheckError] if a git command fails, for example, when the
   /// remote git repository is not reachable due to a network issue.
-  static Future<String> fetchRemoteFrameworkCommitDate(String branch) async {
-    await _removeVersionCheckRemoteIfExists();
+  static Future<String> fetchRemoteFrameworkLatestRevision() async {
+    String revision;
     try {
-      await _run(<String>[
-        'git',
-        'remote',
-        'add',
-        _versionCheckRemote,
-        _flutterGit,
-      ]);
-      await _run(<String>['git', 'fetch', _versionCheckRemote, branch]);
-      return _latestGitCommitDate(
-        branch: '$_versionCheckRemote/$branch',
-        lenient: false,
-      );
+      await _run(<String>['git', 'fetch', '--tags']);
+      // '@{u}' means upstream HEAD
+      revision = await _run(<String>['git', 'rev-parse', '--verify', '@{u}']);
+      return revision;
     } on VersionCheckError catch (error) {
-      if (globals.platform.environment.containsKey('FLUTTER_GIT_URL')) {
-        globals.logger.printError('Warning: the Flutter git upstream was overridden '
-        'by the environment variable FLUTTER_GIT_URL = $_flutterGit');
-      }
       globals.logger.printError(error.toString());
       rethrow;
-    } finally {
-      await _removeVersionCheckRemoteIfExists();
```

Comment:
Since the `_flutterGit` upstream is removed, I have to get the hash and the commit date simultaneously. I couldn't figure out how to do that.

One way could be to return them wrapped in a class, but I think making that change would lead to refactoring.

What do you think?